packages/visual-editing/src/util/dragAndDrop.ts

```diff
@@ -412,38 +424,52 @@ export function handleOverlayDrag(
     }
 
     if (!minimapScaleApplied) {
-      window.removeEventListener('keyup', handleKeyup)
+      clearInterval(rectsInterval)
+      removeListeners()
     }
-
-    window.removeEventListener('mousemove', handleMouseMove)
-    window.removeEventListener('wheel', handleScroll)
-    window.removeEventListener('mouseup', handleMouseUp)
   }
 
   const handleKeyup = (e: KeyboardEvent) => {
     if (e.key === 'Shift' && minimapScaleApplied) {
+      minimapScaleApplied = false
+
       const skeleton = buildPreviewSkeleton(mousePos, element, 1 / scaleFactor)
 
       handler({
         type: 'overlay/dragUpdateSkeleton',
         skeleton,
       })
 
-      resetMinimapWrapperTransform(mousePosInverseTransform.y, scaleTarget, prescaleHeight).then(
-        () => {
-          rects = overlayGroup.map((e) => getRect(e.elements.measureElement))
-
-          minimapScaleApplied = false
-        },
-      )
+      resetMinimapWrapperTransform(mousePosInverseTransform.y, scaleTarget, prescaleHeight)
 
       // cleanup keyup after drag sequence is complete
       if (!mousedown) {
-        window.removeEventListener('keyup', handleKeyup)
+        clearInterval(rectsInterval)
+        removeListeners()
       }
     }
   }
 
+  const handleBlur = () => {
```

Comment:
Hardening to prevent studio focus changes from getting folks "stuck" in minimap views